### DISCUSSION TITLE
Is there any way to create a knowledge base quickly

### DISCUSSION THREAD

**Discussion by gaohuan2015** [explosion/spaCy#9044] [Help: Best practices]
Hi Spacy team

I am training an entity linking model by myself. I use the whole DBpedia dataset to train this model. Before the training procedure, I need to create a knowledge base and import the entities from DBpedia to this kb. Since there are more than 4 million entities and I will get all embeddings for all entities, add_entity method is too slow. Is there any way to add entities to kb faster?

**Comment by polm** [comment]
Do you actually need all four million entries? It seems likely you don't, in which case the fastest thing is probably to filter them down to the ones you actually do need.
